Source/JavaScriptCore/wasm/js/WasmToJS.cpp

```diff
@@ -53,6 +53,25 @@ static void materializeImportJSCell(JIT& jit, unsigned importIndex, GPRReg resul
     jit.loadPtr(JIT::Address(result, Instance::offsetOfImportFunction(importIndex)), result);
 }
 
+static Expected<MacroAssemblerCodeRef<WasmEntryPtrTag>, BindingFailure> handleBadGCTypeIndexUse(VM& vm, JIT& jit, unsigned importIndex)
+{
+    jit.loadWasmContextInstance(GPRInfo::argumentGPR2);
+
+    // Store Callee.
+    jit.loadPtr(CCallHelpers::Address(GPRInfo::argumentGPR2, Instance::offsetOfOwner()), GPRInfo::argumentGPR0);
+    jit.loadPtr(CCallHelpers::Address(GPRInfo::argumentGPR0, JSWebAssemblyInstance::offsetOfModule()), GPRInfo::argumentGPR1);
+    jit.prepareCallOperation(vm);
+    jit.emitPutToCallFrameHeader(GPRInfo::argumentGPR1, CallFrameSlot::callee);
+
+    emitThrowWasmToJSException(jit, GPRInfo::argumentGPR2, Wasm::ExceptionType::InvalidGCTypeUse);
```

Comment:
Why do we assume this is GPR2 and not the pinned instance register?

Reply:
We could use the pinned register here (as in the code in `JSToWasm.cpp`), but since it has to be moved into GPR2 anyway in `emitThrowWasmToJSException` there isn't much difference. (this was based on the old I64 checking code which did the same)